d2-mode.el

```diff
@@ -117,7 +117,7 @@
          (cmd (concat (shell-quote-argument d2-location)
                       " " temp-file
                       " " (org-babel-process-file-name out-file)
-                      " " d2-flags)))
+                      " " (string-join d2-flags " "))))
```

Comment:
Do we need to require `subr-x` to support the `string-join` function?

Reply:
Hey @terlar . I think it's fine breaking compatibility here. I think this is going to be a bump into a new version. Probably 0.2 to signify breaking. Along with #47. Maybe some linting updates as well.  

I'm super swamped today/tomorrow, but I will try to review this over the weekend/later half of the week. Sorry for the delay and thank you again for the contribution. 

string-join is built in Emacs 25. Think we are fine. We can also just add the function directly:

;;;###autoload
(defsubst string-join (strings &optional separator)
"Join all STRINGS using SEPARATOR.
Optional argument SEPARATOR must be a string, a vector, or a list of
characters; nil stands for the empty string."
(declare (pure t) (side-effect-free t))
(mapconcat #'identity strings separator))

I'd prefer to not require a library for something as simple as string-join.

Reply:
sounds good. will get this merged in asap. Something came up this weekend and going to have to delay my review until mid next week. my apologies. I'm looking forward to the review.

Reply:
No worries, then I will have time to update with mapconcat instead

Reply:
https://github.com/andorsk/d2-mode/actions/runs/9472621470/job/26295213297?pr=48#step:5:59 <- string join needs to be added in, per this. I will add a suggestion here for string-join. Lmk if you're ok wth it